include/seqan3/search/fm_index/concept.hpp

```diff
@@ -86,4 +86,226 @@ enum text_layout : bool
     collection
 };
 
+#ifdef SEQAN3_DEPRECATED_310
+// ============================================================================
+//  fm_index_specialisation
+// ============================================================================
+
+/*!\interface seqan3::fm_index_specialisation <>
+ * \brief Concept for unidirectional FM indices.
+ *
+ * This concept defines the interface for unidirectional FM indices.
+ *
+ * \deprecated Use seqan3::detail::template_specialisation_of<typename t::cursor_type, seqan3::fm_index_cursor>
```

Comment:
Mhm, unfortunate that the alternative is in the detail namespace

Reply:
yeah :)

but we decided to remove it entirely without any replacement